conans/client/build/meson.py

```diff
@@ -147,25 +185,37 @@ def configure(self, args=None, defs=None, source_dir=None, build_dir=None,
               "Release": "release"}.get(str(self.build_type), "")
 
         build_type = "--buildtype=%s" % bt
+        cross_option = None
+        environ_append = {"PKG_CONFIG_PATH": pc_paths}
+        is_cross = tools.cross_building(self._conanfile.settings)
+        if is_cross:
+            cross_filename = os.path.join(self.build_dir, "cross_file.txt")
+            cross_option = "--cross-file=%s" % cross_filename
+            self._configure_cross_compile(cross_filename, environ_append)
+
         arg_list = join_arguments([
             "--backend=%s" % self.backend,
             self.flags,
             args_to_string(args),
-            build_type
+            build_type,
+            cross_option,
         ])
         command = 'meson "%s" "%s" %s' % (source_dir, self.build_dir, arg_list)
-        with environment_append({"PKG_CONFIG_PATH": pc_paths}):
-            self._run(command)
+        with environment_append(environ_append):
+            self._run(command, use_auto_tools=not is_cross)
 
     @property
     def _vcvars_needed(self):
         return self._compiler == "Visual Studio" and self.backend == "ninja"
 
-    def _run(self, command):
+    def _run(self, command, use_auto_tools=True):
         with tools.vcvars(self._settings,
                           output=self._conanfile.output) if self._vcvars_needed else tools.no_op():
-            env_build = AutoToolsBuildEnvironment(self._conanfile)
-            with environment_append(env_build.vars):
+            if use_auto_tools:
+                env_build = AutoToolsBuildEnvironment(self._conanfile)
```

Comment:
I'd say ** AutoToolsBuildEnvironment** are needed in both cases, otherwise, we're not setting all the flags correctly